custom_components/ha-glocaltokens/api.py

```diff
@@ -52,43 +52,46 @@ def get_android_id(self):
     def create_url(self, ip, port, api_endpoint):
         url = 'https://{ip}:{port}/{endpoint}'.format(
             ip=ip,
-            port=port,
+            port=str(port),
             endpoint=api_endpoint
         )
         return url
 
-    def get_alarms_and_timers_from(self, ip, port, token, endpoint):
-        url = self.create_url(ip, port, endpoint)
-        HEADERS[HEADER_CAST_LOCAL_AUTH] = token
+    def get_alarms_and_timers_from(self, device, endpoint):
+        url = self.create_url(device[DEVICE_IP], device[DEVICE_PORT], endpoint)
+        _LOGGER.debug("For device {device} - {url}".format(device=device[DEVICE_NAME], url=url))
+        HEADERS[HEADER_CAST_LOCAL_AUTH] = device[TOKEN]
         response = requests.get(url, headers=HEADERS, verify=False, timeout=TIMEOUT) # verify=False is need to avoid SSL security checks. Othervise it will fail to connect
 
-        return response
+        if response.status_code != HTTP_OK:
+            _LOGGER.error("For device {device} - API returned {error}".format(device=device[DEVICE_NAME], error=response.status_code))
+            return False
```

Comment:
Just advice, you could just say `return` here, so your check would look prettier where you call for `get_alarms_and_timers_from`